python3Packages.vllm: 0.3.3->0.5.2

cfhammill opened the pull request:
## Description of changes

Updates vllm. Requires two new python packages, lm-format-enforcer and prometheus-fastapi-instrumentator, this PR adds both.

I was able to get the cuda version to build on a slightly older revision of nixpkgs. Would have needed to rebuild the entire python/pytorch/cuda stack to try against master.  Compiling vllm alone takes > 1h so this was slow going. Also have not built against rocm.

@CertainLach you mentioned that this need a special triton? It compiles ok for me with the version I fixed in https://github.com/NixOS/nixpkgs/pull/325843.

It compiles and can be imported, but since tests have not been enabled (and likely wouldn't work anyway) unsure if this works as expected.

CC @SomeoneSerge @ConnorBaker @samuela 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

CertainLach commented:
> @CertainLach you mentioned that this need a special triton? It compiles ok for me with the version I fixed in https://github.com/NixOS/nixpkgs/pull/325843.

It was failing at runtime without new triton for me, I still maintain fork of nixpkgs with updated triton because of that.
Not sure if they have fixed this version requirement now, will check it on this PR later

samuela commented on pkgs/development/python-modules/lm-format-enforcer/default.nix:
```diff
@@ -0,0 +1,41 @@
+buildPythonPackage rec {
+  pname = "lm-format-enforcer";
+  version = "0.10.4";
+  format = "pyproject";
```
i can never remember... is the latest guidance `pyproject = true;` or ` format = "pyproject";`?

SomeoneSerge replied:
`pyproject = true`

samuela commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -38,18 +48,24 @@
+  cutlass = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "cutlass";
+    rev = "v3.5.0";
```
```suggestion
    rev = "refs/tags/v3.5.0";
```

samuela commented on pkgs/development/python-modules/lm-format-enforcer/default.nix:
```diff
@@ -0,0 +1,41 @@
+  src = fetchFromGitHub {
+    owner = "noamgat";
+    repo = pname;
+    rev = "v${version}";
```
```suggestion
    rev = "refs/tags/v${version}";
```

samuela commented on pkgs/development/python-modules/prometheus-fastapi-instrumentator/default.nix:
```diff
@@ -0,0 +1,39 @@
+  src = fetchFromGitHub {
+    owner = "trallnag";
+    repo = pname;
+    rev = "v${version}";
```
```suggestion
    rev = "refs/tags/v${version}";
```

samuela commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -62,51 +78,42 @@ buildPythonPackage rec {
     torch
     wheel
     which
+    cmake
+    ninja
+    pythonRelaxDepsHook #unclear why this is needed, but it seems to be
```
why not maintain alphabetical order?

samuela replied:
`pythonRelaxDepsHook` is necessary for the `pythonRelaxDeps = true;` you're using below. see https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonrelaxdepshook-using-pythonrelaxdepshook for more info

cfhammill replied:
ahh I was under the impression that explicitly importing the hook wasn't needed anymore. Removed the comment and realphabetized.

SomeoneSerge replied:
> pythonRelaxDepsHook necessary

Should be propagated by `buildPythonPackage` nowadays no?

samuela commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -62,51 +78,42 @@ buildPythonPackage rec {
-        cuda_cccl # <thrust/*>
-        libcusparse # cusparse.h
-        libcublas # cublas_v2.h
+        cudatoolkit
```
we're really trying hard to get off of `cudatoolkit` and onto the redist packages instead. using both in the same derivation is a little wonky

SomeoneSerge replied:
Yes please let's not create more consumers for the merged toolkit

cfhammill replied:
so I'm not sure which of the redist packages are needed here, nor how to tell vllm where to find them. I suspect I would need to do a symlinkJoin of all the redist packages and use that as CUDA_HOME? I'm not sure when I will have time to figure this out so if anyone has the drive to take a crack at it, it would get done faster.

SomeoneSerge replied:
It seems to have worked without the symlinkJoin before,so it should

SomeoneSerge replied:
> CUDA_HOME

Often (and apparently in case of vllm) has no real effect, so we just put something like nvcc or cudart to bypass useless upstream checks

cfhammill replied:
ok thanks, that's really helpful - looks like it was missing a few packages, I think I got it to build without the toolkit or symlinkjoin. Will know in 45 minutes.

cfhammill replied:
it built successfully

samuela commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -145,7 +159,11 @@ buildPythonPackage rec {
+
+  pythonRelaxDeps = true;
+
+  stdenv = if cudaSupport then cudaPackages.backendStdenv else stdenv;
```
shouldn't this be 
```
if cudaSupport then cudaPackages.backendStdenv else stdenv_pkg;
```
?

SomeoneSerge replied:
Also remember that there's an alternative to `let ... in`, which is to rewrite the top-level args as `{ ... }@args:` and use `if cudaSupport then ... else args.stdenv`

cfhammill replied:
sorry this was a merge fail, and great idea, will switch to using `@args`

SomeoneSerge commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -62,51 +78,42 @@ buildPythonPackage rec {
-        libcusparse # cusparse.h
-        libcublas # cublas_v2.h
+        cudatoolkit
+        cuda_cudart.dev # cuda_runtime.h, -lcudart
```
.dev is selected automatically; .lib is propagated by .dev

SomeoneSerge commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -145,7 +159,11 @@ buildPythonPackage rec {
+
+  pythonRelaxDeps = true;
+
+  stdenv = if cudaSupport then cudaPackages.backendStdenv else stdenv;
```
I wonder if it's more visible e.g. above nativeBuildInputs, under src/patches

SomeoneSerge commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -62,51 +78,42 @@ buildPythonPackage rec {
     which
+    cmake
+    ninja
+    pythonRelaxDepsHook #unclear why this is needed, but it seems to be
```
So did this actually make a difference? If it did and you keep this line, add a space after #

cfhammill replied:
yeah it did. I'll re-add the comment

cfhammill replied:
we encountered relaxDeps mysteriously not working for vllm previously here: https://github.com/NixOS/nixpkgs/pull/326145#discussion_r1672795111

cfhammill commented:
I think I've addressed everything, it should be good to go @samuela @SomeoneSerge @CertainLach.

Let me know if you want me to squash the commits.

SomeoneSerge commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -90,32 +92,31 @@ buildPythonPackage rec {
     ninja
     packaging
-    pythonRelaxDepsHook
+    pythonRelaxDepsHook # not sure why this is needed, but it is
```
```suggestion
pythonRelaxDepsHook # `pythonRelaxDeps = true` doesn't work otherwise
```

SomeoneSerge commented:
Result of `nixpkgs-review pr 328258 --extra-nixpkgs-config '{
  cudaSupport = true;
  cudaCapabilities = [
    "7.0"
    "8.0"
    "8.6"
  ];
  cudaEnableForwardCompat = false;
  allowUnfreePredicate =
    p:
    builtins.all (
      license:
      license.free
      || builtins.elem license.shortName [
        "CUDA EULA"
        "cuDNN EULA"
        "cuTENSOR EULA"
        "NVidia OptiX EULA"
      ]
    ) (if builtins.isList p.meta.license then p.meta.license else [ p.meta.license ]);
  permittedInsecurePackages = [
    "electron-24.8.6"
    "zotero-6.0.27"
  ];
}'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.vllm</li>
    <li>python311Packages.vllm.dist</li>
    <li>python312Packages.vllm</li>
    <li>python312Packages.vllm.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.lm-format-enforcer</li>
    <li>python311Packages.lm-format-enforcer.dist</li>
    <li>python311Packages.prometheus-fastapi-instrumentator</li>
    <li>python311Packages.prometheus-fastapi-instrumentator.dist</li>
    <li>python312Packages.lm-format-enforcer</li>
    <li>python312Packages.lm-format-enforcer.dist</li>
    <li>python312Packages.prometheus-fastapi-instrumentator</li>
    <li>python312Packages.prometheus-fastapi-instrumentator.dist</li>
  </ul>
</details>

cfhammill commented:
thanks @SomeoneSerge, any idea why the vllm build is marked broken? Your config should have had cuda enabled

SomeoneSerge commented:
```
┃ 
┃        error: Package ‘python3.12-outlines-0.0.45’ in /home/ss/.cache/nixpkgs-review/pr-328258-1/nixpkgs/pkgs/development/python-modules/outlines/default.nix:56 is marked as broken, refusing to evaluate.
┃ 
┃        a) To temporarily allow broken packages, you can use an environment variable
┃           for a single invocation of the nix tools.
┃ 

```


broken on master (Ofborg rebases PRs on master before running any checks)

cfhammill commented:
outlines was broken in ff7d46d1202432dff5ea8797d5b893eff3bd434a by an untested version bump, and marked as broken a few days ago when someone noticed.

I fixed and tacked it on to this PR.

samuela commented on pkgs/development/python-modules/lm-format-enforcer/default.nix:
```diff
@@ -0,0 +1,37 @@
+
+  nativeBuildInputs = [ poetry-core ];
+  propagatedBuildInputs = [
+    pydantic
+    interegular
+    pyyaml
```
nit: alphabetize

samuela commented on pkgs/development/python-modules/prometheus-fastapi-instrumentator/default.nix:
```diff
@@ -0,0 +1,36 @@
+  nativeBuildInputs = [ poetry-core ];
+
+  propagatedBuildInputs = [
+    starlette
+    prometheus-client
```
nit: alphabetize

cfhammill commented:
A new version of vllm has been released, seems like a small one so hopefully no need to modify the build other than bump version and hash. Trying now.

SomeoneSerge commented:
Result of `nixpkgs-review pr 328258 --extra-nixpkgs-config '{ cudaSupport = true; cudaCapabilities = [ "7.0" "8.0" "8.6" ]; cudaEnableForwardCompat = false; allowUnfreePredicate = p: builtins.all ( license: license.free || builtins.elem license.shortName [ "CUDA EULA" "cuDNN EULA" "cuTENSOR EULA" "NVidia OptiX EULA" ] ) (if builtins.isList p.meta.license then p.meta.license else [ p.meta.license ]); permittedInsecurePackages = [ "electron-24.8.6" "zotero-6.0.27" ]; }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.outlines</li>
    <li>python311Packages.outlines.dist</li>
    <li>python311Packages.vllm</li>
    <li>python311Packages.vllm.dist</li>
    <li>python312Packages.outlines</li>
    <li>python312Packages.outlines.dist</li>
    <li>python312Packages.vllm</li>
    <li>python312Packages.vllm.dist</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python311Packages.lm-format-enforcer</li>
    <li>python311Packages.lm-format-enforcer.dist</li>
    <li>python311Packages.prometheus-fastapi-instrumentator</li>
    <li>python311Packages.prometheus-fastapi-instrumentator.dist</li>
    <li>python311Packages.pyairports</li>
    <li>python311Packages.pyairports.dist</li>
    <li>python312Packages.lm-format-enforcer</li>
    <li>python312Packages.lm-format-enforcer.dist</li>
    <li>python312Packages.prometheus-fastapi-instrumentator</li>
    <li>python312Packages.prometheus-fastapi-instrumentator.dist</li>
    <li>python312Packages.pyairports</li>
    <li>python312Packages.pyairports.dist</li>
  </ul>
</details>

cfhammill commented:
sorry, removing the broken annotation on outlines (I'm working across two branches [a semi-stable one that builds and works for me and includes a vllm patch for MIG supper, and this one] and I'm evidently failing to merge/cherrypick properly, apologies!)

SomeoneSerge commented:
> evidently failing to merge/cherrypick properly, apologies!)

No worries. It's just

```
git fetch origin # assuming origin refers to git@github.com:NixOS/nixpkgs
git rebase -i origin/master
```

Also you might want to look into lazygit

cfhammill commented:
Thanks @SomeoneSerge. Turns out this was not the issue I thought, this PR does have the broken annotation removed, not sure how it got reintroduced, checking now.

bcdarwin commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -34,65 +44,61 @@
+    cmake
     ninja
     packaging
+    pythonRelaxDepsHook # `pythonRelaxDeps = true` doesn't work otherwise
```
This comment is unnecessary (common knowledge).

cfhammill replied:
see discussion with @SomeoneSerge, it should be imported automatically by `buildPythonPackage`, so this is a surprise.

cfhammill replied:
https://github.com/NixOS/nixpkgs/pull/328258#discussion_r1685324728

bcdarwin replied:
First I've heard of this - there are 355 uses of pythonRelaxDeps/pythonRemoveDeps and only 2 don't import it (presumably broken).

cfhammill replied:
ok - I was tricked by a treewide removal of pythonRelaxDepsHook `treewide: clean up pythonRelaxDepsHook references`, but those packages didn't use relaxDeps. Will remove the comment.

bcdarwin commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -34,65 +44,61 @@
   rocmPackages ? { },
   gpuTargets ? [ ],
-}:
+}@args:
```
would suggest using let-binding `stdenv` appropriately to avoid the shadowing instead of capturing via `@`

cfhammill replied:
that was the original approach see https://github.com/NixOS/nixpkgs/pull/328258#discussion_r1684482396

SomeoneSerge replied:
We don't `inherit` because we want to choose between `stdenv` and `cudaPackages.backendStdenv`

bcdarwin commented on pkgs/development/python-modules/pyairports/default.nix:
```diff
@@ -0,0 +1,27 @@
+
+  meta = with lib; {
+    description = "pyairports is a package which enables airport lookup by 3-letter IATA code.";
+    #changelog = ""; # package does not have one
```
can remove this commented `changelog`

bcdarwin commented on pkgs/development/python-modules/pyairports/default.nix:
```diff
@@ -0,0 +1,27 @@
+buildPythonPackage rec {
+  pname = "pyairports";
+  version = "2.1.1";
+  format = "pyproject";
```
`pyproject = true` is preferred

cfhammill replied:
good point, will switch.

bcdarwin commented on pkgs/development/python-modules/pyairports/default.nix:
```diff
@@ -0,0 +1,27 @@
+    hash = "sha256-PWCnJ/zk2oG5xjk+qK4LM9Z7N+zjRN/8hj90njrWK80=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
```
`build-system` is preferred to `nativeBuildInputs` for pythonPackages

SomeoneSerge replied:
(@cfhammill to clarify, it's preferred for python dependencies, but not for e.g. hooks)

bcdarwin commented on pkgs/development/python-modules/pyairports/default.nix:
```diff
@@ -0,0 +1,27 @@
+  };
+
+  nativeBuildInputs = [ setuptools ];
+
```
please at least set `pythonImportsCheck`, and also set `doCheck = false` if there are no tests.

bcdarwin reviewed (changes requested):
Made some largely cosmetic suggestions, but please do squash into commits for vllm, outlines, and pyairports.

cfhammill commented:
will squash shortly, thanks @bcdarwin

bcdarwin reviewed (approved):
Looking good to me.

SomeoneSerge commented on pkgs/development/python-modules/lm-format-enforcer/default.nix:
```diff
@@ -0,0 +1,42 @@
+    interegular
+    pydantic
+    pyyaml
+  ];
```
`dependencies = ...`

cfhammill replied:
why is this preferred? I copied the style from outlines.

bcdarwin replied:
pythonPackages now prefers `build-system` and `dependencies` instead of `nativeBuildInputs` and `propagatedBuildInputs` to match Python native packaging (e.g. pyproject.toml) style.

SomeoneSerge replied:
Also we're hoping to stop using references (nix-support/propagated-build-inputs) for python deps eventually, to reduce # rebuilds

SomeoneSerge commented on pkgs/development/python-modules/outlines/default.nix:
```diff
@@ -48,16 +52,19 @@ buildPythonPackage rec {
-  pythonImportsCheck = [ "outlines" ];
+  checkPhase = ''
+    export HOME=$(mktemp -d)
+    python3 -c 'import outlines'
```
This is `pythonImportsCheck` with `preCheck = ''export HOME=..''`

cfhammill replied:
I tried that first and it did not work, I don't think `preCheck` applies to the import check.

SomeoneSerge replied:
preInstallCheck maybe

cfhammill replied:
I think given the rest is good to go that we can leave this for now as it works, it can be improved in a future PR.

SomeoneSerge commented on pkgs/development/python-modules/lm-format-enforcer/default.nix:
```diff
@@ -0,0 +1,42 @@
+
+  doCheck = false; # most tests require internet access
+
+  pythonImportChecks = [ "lmformatenforcer" ];
```
*Check singular; verify `nix log` for `lmformatenforcer` actually being imported

SomeoneSerge replied:
Imports plural:)

SomeoneSerge replied:
in general, grep nixpkgs for examples

cfhammill replied:
oops, fixing.

cfhammill replied:
I blame nix for this one, because the inputs didn't change this didn't rebuild when I test it locally.

SomeoneSerge commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -123,29 +131,38 @@ buildPythonPackage rec {
     ];
 
-  stdenv = if cudaSupport then cudaPackages.backendStdenv else stdenv_pkg;
+  dontUseCmakeConfigure = true;
```
Nit: this should go before `configurePhase` and respective hooks; side note: this `default.nix` also puts `*Inputs` after configure/build phases which is unconventional for nixpkgs and so less convenient to read

cfhammill replied:
this default.nix doesn't have a configurePhase, should it go above the Inputs?

SomeoneSerge replied:
I think we oscillate somewhere around

```
pname
version
pyproject
disabled
src
patches
postPatch
[native]BuildInputs
dontUseCmakeConfigure
configurePhase
buildPhase
checkPhase
installPhase
postInstallPhase # the real install phase in case of python
```

cfhammill replied:
looking at this list I still don't really see what's out of place, we have
```
pname
version
src
stdenv
<env var>
postPatch
preBuild
*buildInputs
dontUseCmakeConfigure
relaxDeps
importsCheck
meta
```

is it that preBuild belongs before the (missing) buildPhase? So between dontUseCmakeConfigure and relaxDeps?

SomeoneSerge commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -34,65 +44,61 @@
+    cmake
     ninja
     packaging
+    pythonRelaxDepsHook
     setuptools
     torch
```
Side note: torch in nativeBuildInputs looks wrong

cfhammill replied:
yeah, not sure why that was put there.

bcdarwin replied:
Maybe one of those 'let the user bring their own `torch`' ideas a la the current `torchmetrics`?

SomeoneSerge replied:
> Maybe one of those 'let the user bring their own torch' ideas a la the current torchmetrics?

I rather meant that vllm likely needs the host platform's torch, not the build platform's

cfhammill replied:
ok - I think this torch can be removed, torch is already in propagatedBuildInputs, shouldn't be needed twice.

SomeoneSerge commented:
Result of `nixpkgs-review pr 328258 --extra-nixpkgs-config '{ cudaSupport = true; cudaCapabilities = [ "7.0" "8.0" "8.6" ]; cudaEnableForwardCompat = false; allowUnfreePredicate = p: builtins.all ( license: license.free || builtins.elem license.shortName [ "CUDA EULA" "cuDNN EULA" "cuTENSOR EULA" "NVidia OptiX EULA" ] ) (if builtins.isList p.meta.license then p.meta.license else [ p.meta.license ]); permittedInsecurePackages = [ "electron-24.8.6" "zotero-6.0.27" ]; }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.vllm</li>
    <li>python312Packages.vllm.dist</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>python311Packages.lm-format-enforcer</li>
    <li>python311Packages.lm-format-enforcer.dist</li>
    <li>python311Packages.outlines</li>
    <li>python311Packages.outlines.dist</li>
    <li>python311Packages.pyairports</li>
    <li>python311Packages.pyairports.dist</li>
    <li>python311Packages.vllm</li>
    <li>python311Packages.vllm.dist</li>
    <li>python312Packages.lm-format-enforcer</li>
    <li>python312Packages.lm-format-enforcer.dist</li>
    <li>python312Packages.outlines</li>
    <li>python312Packages.outlines.dist</li>
    <li>python312Packages.pyairports</li>
    <li>python312Packages.pyairports.dist</li>
  </ul>
</details>

cfhammill commented:
@SomeoneSerge do you have the build logs for python312Packages.vllm?

SomeoneSerge commented:
I had to re-run nixpkgs-review because logs got GC-ed, but here: https://gist.github.com/SomeoneSerge/6a3344368c146171160a92d6865ffd57

TLDR: upstream marked vllm as only supporting up to 3.11; I highly doubt that was intentional though

SomeoneSerge commented:
https://github.com/vllm-project/vllm/blob/af647fb8b3ea9d910f7d1bc104af8986d048a8e2/CMakeLists.txt#L17

SomeoneSerge commented:
So uh I'm not a vllm user, you probably know better: is there value in providing `python311Packages.vllm` but not `python3Packages.vllm`? I'm content to merge the PR (as is plus the `disabled = ...` attribute) since it's upstream who decided not to support 3.12

cfhammill commented:
I'm using python311 so not having 312 doesn't bother me personally, but it looks like there isn't a good reason it's disabled.
https://github.com/vllm-project/vllm/issues/6877
other than excessively tight version bounds. I'd be ok with either disabling 312 or patching the cmake file to allow 312 and hoping that works.

SomeoneSerge committed:
python311Packages.vllm: postPatch -> patches

Co-authored-by: cfhammill <christopher.hammill@unityhealth.to>

Commit:
python3Packages.pyairports: init at 2.1.1

Commit:
python311Packages.vllm: 0.3.2->0.5.2

vllm remove comment, minor version bump

use refs/tags and format=pyproject

use @args and realphabetize inputs

move stdenv, mv cuda_home, add/clean cuda deps, restore comment

nixfmt rfc style

remove stale comment

SomeoneSerge committed:
python311Packages.vllm: reorder, filter, rename attrs as conventional

pep-517 attr names, no native torch, phase order

Co-authored-by: cfhammill <christopher.hammill@unityhealth.to>

Commit:
python3Packages.lm-format-enforcer: init at 0.10.4

SomeoneSerge committed:
python312Packages.vllm: unbreak in vanilla nixpkgs (default to rocm)

Co-authored-by: cfhammill <christopher.hammill@unityhealth.to>

SomeoneSerge committed:
python312Packages.vllm: allow building for 3.12

Co-authored-by: cfhammill <christopher.hammill@unityhealth.to>

SomeoneSerge committed:
python311Packages.vllm: export in preBuild -> env

Co-authored-by: cfhammill <christopher.hammill@unityhealth.to>

SomeoneSerge commented:
@cfhammill I just added the stuff I said was optional, idk why; the python 3.12 build... doesn't immediately fail (it's been running for 40' now) so I intend to leave it at that; I'll check on Ofborg tomorrow and probably just merge, but please do run some checks and skim over the extra stuff I pushed that it makes sense. I don't use vllm so I can't reason much about this

cfhammill commented on pkgs/development/python-modules/vllm/0002-setup.py-nix-support-respect-cmakeFlags.patch:
```diff
@@ -0,0 +1,40 @@
++    with open(os.environ["NIX_ATTRS_JSON_FILE"], "r") as f:
++        NIX_ATTRS = json.load(f)
++else:
++    NIX_ATTRS = { "cmakeFlags": os.environ.get("cmakeFlags", "").split() }
```
if this makes vllm honour cmakeFlags, then we could probably remove my `substituteInPlace` and provide as a cmakeFlag. Not critical to change obviously.

cfhammill replied:
oops you already patched it!

cfhammill commented:
I'm happy with this! Thanks @SomeoneSerge! Merge when ready.

SomeoneSerge commented:
Thanks @cfhammill
